uportal-war/src/main/java/org/apereo/portal/layout/dlm/remoting/UpdatePreferencesServlet.java

```diff
@@ -231,6 +232,65 @@ public ModelAndView removeElement(HttpServletRequest request, HttpServletRespons
     }
 
     /**
+     * Remove the first element with the provided fname from the layout.
+     *
+     * @param request
+     * @param response
+     * @param fname
+     * @return
+     * @throws IOException
+     */
+    @RequestMapping(method = RequestMethod.POST, params = "action=removeByFName")
+    public ModelAndView removeByFName(HttpServletRequest request, 
+            HttpServletResponse response,
+            @RequestParam(value = "fname", required = true) String fname)
+            throws IOException {
+
+        IUserInstance ui = userInstanceManager.getUserInstance(request);
+        IPerson per = getPerson(ui, response);
+
+        UserPreferencesManager upm = (UserPreferencesManager) ui.getPreferencesManager();
+        IUserLayoutManager ulm = upm.getUserLayoutManager();
+
+        try {
+            String elementId = ulm.getUserLayout().findNodeId(new PortletSubscribeIdResolver(fname));
+            if (elementId != null
+                    && elementId.startsWith(Constants.FRAGMENT_ID_USER_PREFIX)
+                    && ulm.getNode(elementId)
+                            instanceof org.apereo.portal.layout.node.UserLayoutFolderDescription) {
+
+                removeSubscription(per, elementId, ulm);
+
+            } else {
+                // Delete the requested element node.  This code is the same for
+                // all node types, so we can just have a generic action.
+                if (!ulm.deleteNode(elementId)) {
+                    logger.info(
+                            "Failed to remove element ID {} from layout root folder ID {}, delete node returned false",
+                            elementId,
+                            ulm.getRootFolderId());
+                    response.sendError(HttpServletResponse.SC_FORBIDDEN);
```

Comment:
With the exception of one line, this entire function is copied from removeElement.  I agree with you that this should be a SC_BAD_REQUEST, and in uPortal 4.2 it was, however that was changed in this commit: https://github.com/Jasig/uPortal/commit/6a26a16507f116c2f907e31a1d9434f116366bec#diff-ef2d279bd054b445090ea5dc95d41d2aR210
driven by this JIRA: https://issues.jasig.org/browse/UP-4549

It's not clear to me why SC_FORBIDDEN is necessary here.